File: python/pagesnap/storage/storage_provider.py
import abc

class StorageProvider(abc.ABC):
    def __init__(self, options: dict):
        self.options = options

    @abc.abstractmethod
    async def save(self, file_name: str, data: bytes) -> str:
        """Saves the image data."""
        raise NotImplementedError

    async def cleanup(self):
        """Performs cleanup of expired assets."""
        print(f"Cleanup not implemented for {self.__class__.__name__}")
        return

File: packages/core/src/storage/StorageProvider.js
// Base class for all storage providers
export default class StorageProvider {
  constructor(options) {
    this.options = options;
  }

  /**
   * Saves the image data.
   * @param {string} fileName - The name of the file to save.
   * @param {Buffer} data - The image data buffer.
   * @returns {Promise<string>} The path or identifier for the saved file.
   */
  async save(fileName, data) {
    throw new Error('StorageProvider#save must be implemented by subclasses');
  }

  /**
   * Performs cleanup of expired assets.
   * @returns {Promise<void>}
   */
  async cleanup() {
    // Optional: implement in subclasses if TTL is supported
    console.log(`Cleanup not implemented for ${this.constructor.name}`);
  }
}